strawberryfields/backends/tfbackend/backend.py

```diff
@@ -361,17 +322,8 @@ def measure_homodyne(self, phi, mode, shots=1, select=None, **kwargs):
 
     def is_vacuum(self, tol=0.0, **kwargs):
         with tf.name_scope("Is_vacuum"):
-            with self.circuit.graph.as_default():
-                vac_elem = self.circuit.vacuum_element()
-                if "eval" in kwargs and kwargs["eval"] is False:
-                    v = vac_elem
-                else:
-                    sess = tf.Session()
-                    v = sess.run(vac_elem)
-                    sess.close()
-
-            result = np.abs(v - 1) <= tol
-        return result
+            vac_elem = self.circuit.vacuum_element()
+            return np.abs(vac_elem - 1) <= tol
```

Comment:
How come? 😆

Reply:
calling `np.something()` on a tensorflow tensor

Reply:
I'm assuming duck typing saves us here

Reply:
Yep, they've been very clever here (also coupled with the fact that NumPy functions make huge use of duck typing, and very rarely check if the input is an `ndarray`). So as long as `tf.tensor` looks like an iterable (which it does), most NumPy functions will work 🙂

Reply:
Although, there is a slight subtlety here. While `tf.tensor` objects _are_ iterables, `tf.Variable` objects aren't! Which causes a weird edge case, where

```python
>>> x = tf.Variable([0.0, 1.0, 2.0])
>>> list(x) # fails since Variables are not iterable
>>> list(x*1.0) # works since multiplying a variable by a scalar casts it to a pure tensor which is iterable
[<tf.Tensor: id=17, shape=(), dtype=int32, numpy=0>,
 <tf.Tensor: id=21, shape=(), dtype=int32, numpy=1>,
 <tf.Tensor: id=25, shape=(), dtype=int32, numpy=2>]
```

Reply:
So NumPy functions will work great on `tf.constant`, `tf.tensor` objects, but will usually fail directly on `tf.Variable` objects 😆